git-branchless-lib/src/git/oid.rs

```diff
@@ -154,10 +154,18 @@ impl FromStr for MaybeZeroOid {
     type Err = eyre::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s.parse() {
+        let oid = s
+            // First try parsing as a hex commit hash.
+            .parse()
+            // If that doesn't work, attempt parsing as a commit ref.
+            .or_else(|_| git2::Oid::hash_object(git2::ObjectType::Commit, s.as_bytes()));
```

Comment:
hash_object is wrong; we need to resolve the reference name. I think this requires a repository object. Working on a draft for this, but will probably require some architectural guidance, since I'm pretty new to the codebase.

Reply:
Ok, this has been improved.